supervisor/addons/addon.py

```diff
@@ -1456,10 +1462,16 @@ def _restore_data():
                     elif self.addon_config_used:
                         self.path_config.mkdir()
 
+                def check_data_config() -> tuple[bool, bool]:
+                    return (self.path_data.is_dir(), self.path_config.is_dir())
+
                 _LOGGER.info("Restoring data and config for addon %s", self.slug)
-                if self.path_data.is_dir():
+                data_is_dir, config_is_dir = await self.sys_run_in_executor(
+                    check_data_config
+                )
+                if data_is_dir:
                     await remove_data(self.path_data)
```

Comment:
From what I can tell, `remove_data` should be fine if data dir doesn't exist. So we could run this unconditionally :thinking:

Reply:
Ok I'll check that. I was trying to make minimal logic changes to this one to keep bugs from getting in so I mostly just replaced as is.